packages/pwa-kit-react-sdk/src/ssr/universal/hooks/index.js

```diff
@@ -72,6 +73,36 @@ export const useOrigin = ({fromXForwardedHeader = false}) => {
 }
 
 /**
+ * Blocks the navigation to run a provided function whenever there is a new page being navigated to
+ * The function must return false to unblock the navigation
+ *
+ * @param {function} func
+ * @returns {boolean} State that is set to true during blocking, false otherwise.
+ */
+export const useBlockNavigation = (func) => {
+    const {block, push, location} = useHistory()
+    const lastLocation = useRef()
+    const [isBlocked, setIsBlocked] = useState(false)
+    const funcRef = useRef(func)
+
+    useEffect(() => {
+        const unblock = block(async (location, action) => {
+            if (location?.pathname !== lastLocation.current?.pathname && funcRef.current) {
+                lastLocation.current = location
+                setIsBlocked(true)
+                await funcRef.current(location, action)
+                setIsBlocked(false)
+                unblock()
+                push(location?.pathname)
+            }
+        })
+        return () => unblock()
+    }, [location])
```

Comment:
It turns out that `block` is meant to be called sparingly. It's meant to register a callback whenever a navigation change is about to happen. Like an onClick event handler, you call it once to register.

So then with `unblock`, you can read it as "unregister the block handler". And similarly, you call that once (in the returned cleanup function).

```suggestion
    useEffect(() => {
        const unblock = block(async (location, action) => {
            if (location?.pathname !== lastLocation.current?.pathname && funcRef.current) {
                lastLocation.current = location
                setIsBlocked(true)
                await funcRef.current(location, action)
                setIsBlocked(false)
                // unblock()
                push(location?.pathname)
            }
        })
        return () => unblock()
    }, []) // Make sure to register the navigation block handler _once_
```